system/storage/region.toit

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2023 Toitware ApS.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; version
+// 2.1 only.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// The license can be found in the file `LICENSE` in the top level
+// directory of this repository.
+
+import .storage show StorageServiceProvider
+
+import binary show LITTLE_ENDIAN
+import encoding.tison
+import system.services show ServiceResource
+import system.storage show Region
+import uuid
+
+import ..flash.allocation
+import ..flash.registry
+
+class FlashRegionResource extends ServiceResource:
+  static NAMESPACE ::= "flash:region"
+
+  client_/int
+  handle_/int? := null
+
+  constructor provider/StorageServiceProvider .client_ --offset/int --size/int:
+    super provider client_
+    try:
+      handle := serialize_for_rpc
+      flash_grant_access_ client_ handle offset size
+      handle_ = handle
+    finally: | is_exception _ |
+      if is_exception: close
+
+  revoke -> none:
+    if not handle_: return
+    flash_revoke_access_ client_ handle_
+    handle_ = null
+
+  on_closed -> none:
+    revoke
+
+  static open provider/StorageServiceProvider client/int -> List
+      --path/string
+      --capacity/int?:
+    registry := provider.registry
+    id := uuid.uuid5 NAMESPACE path
+    allocation := find_allocation_ registry --id=id --if_absent=:
+      if not capacity: throw "FILE_NOT_FOUND"
+      new_allocation_ registry --id=id --path=path --size=capacity + FLASH_REGISTRY_PAGE_SIZE
```

Comment:
Add a comment explaining why we need the extra FLASH_REGISTRY_PAGE_SIZE

Reply:
Done.